File: examples/location/search/node/searchForLocations.js
const { LocationService } = require("m3o/location");

// Search for entities in a given radius
async function searchForLocations() {
  let locationService = new LocationService(process.env.MICRO_API_TOKEN);
  let rsp = await locationService.search({
    center: {
      latitude: 51.511061,
      longitude: -0.120022,
    },
    numEntities: 10,
    radius: 100,
    type: "bike",
  });
  console.log(rsp);
}

searchForLocations();

File: examples/go/location/search/searchForLocations.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/location"
)

// Search for entities in a given radius
func SearchForLocations() {
	locationService := location.NewLocationService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := locationService.Search(&location.SearchRequest{
		Center: &location.Point{
			Latitude:  51.511061,
			Longitude: -0.120022,
		},
		NumEntities: 10,
		Radius:      100,
		Type:        "bike",
	})
	fmt.Println(rsp, err)
}
